app/controllers/tool_groups_controller.rb

```diff
@@ -2,7 +2,13 @@ class ToolGroupsController < ApplicationController
   before_action :authorize!
 
   def index
-    render json: tool_groups_ordered_by_name, status: :ok
+    include = {}
+    fields = {}
+
+    include = params[:include]&.split(",") if params[:include]
+    fields = get_fields(params[:fields]) if params[:fields]
```

Comment:
I'm guessing you could use the [`fields_param`](https://github.com/CruGlobal/mobile-content-api/blob/master/app/controllers/concerns/fields.rb#L4) function here and not need to implement a separate get_fields() function.

I haven't looked into the pre-existing vs your implementation to know which one would be more accurate/safe

Reply:
That is right! I think the problem was just in the way the params were passed in the specs and that was not working with the  `fields_param` . Now it works well with the pre existent `fields_param` function